pkg/scheduler/frameworkext/eventhandlers/reservation_handler.go

```diff
@@ -327,43 +326,59 @@ func updateReservationInCache(sched frameworkext.Scheduler, oldObj, newObj inter
 	if err := sched.GetCache().UpdatePod(oldReservePod, newReservePod); err != nil {
 		klog.Errorf("scheduler cache UpdatePod failed for reservation, old %s, new %s, err: %v", klog.KObj(oldR), klog.KObj(newR), err)
 	}
-	sched.GetSchedulingQueue().AssignedPodAdded(newReservePod)
+	sched.GetSchedulingQueue().AssignedPodUpdated(newReservePod)
 }
 
 func deleteReservationFromCache(sched frameworkext.Scheduler, obj interface{}) {
-	var r *schedulingv1alpha1.Reservation
-	switch t := obj.(type) {
-	case *schedulingv1alpha1.Reservation:
-		r = t
-	case cache.DeletedFinalStateUnknown:
-		var ok bool
-		r, ok = t.Obj.(*schedulingv1alpha1.Reservation)
-		if !ok {
-			klog.Errorf("deleteReservationFromCache failed, cannot convert to *schedulingv1alpha1.Reservation, obj %T", t.Obj)
-			return
-		}
-	default:
+	r := toReservation(obj)
+	if r == nil {
 		klog.Errorf("deleteReservationFromCache failed, cannot convert to *schedulingv1alpha1.Reservation, obj %T", obj)
 		return
 	}
 	klog.V(3).InfoS("Delete event for scheduled reservation", "reservation", klog.KObj(r))
 
+	if r.Status.NodeName == "" {
+		return
+	}
+
 	// delete pod cache and trigger pod deleted event for scheduling queue
 	err := reservationutil.ValidateReservation(r)
 	if err != nil {
 		klog.Errorf("deleteReservationFromCache failed, invalid reservation, err: %v", err)
 		return
 	}
+
+	reservationCache := reservation.GetReservationCache()
+	rInfo := reservationCache.DeleteReservation(r)
+	if rInfo == nil {
+		klog.Warningf("The impossible happened. Missing ReservationInfo in ReservationCache, reservation: %v", klog.KObj(r))
+		return
+	}
+
 	reservePod := reservationutil.NewReservePod(r)
-	if err := sched.GetCache().RemovePod(reservePod); err != nil {
-		klog.Errorf("scheduler cache RemovePod failed for reservation, reservation %s, err: %v", klog.KObj(r), err)
+	if _, err = sched.GetCache().GetPod(reservePod); err == nil {
```

Comment:
existPod, err = sched.GetCache().GetPod(reservePod)
here may remove the usedports on the existPod  in the cache. not the reservePod.
because in  RemovePod method , it use the pod which in the cache
![image](https://github.com/koordinator-sh/koordinator/assets/21192363/667cecce-9214-4ac2-afed-431d5321b8ae)

Reply:
Yes.. So before remove, must update the pod first..